src/player.cpp

```diff
@@ -5663,12 +5663,12 @@ void player::try_to_sleep( const time_duration &dur )
         plantsleep = true;
         if( ( ter_at_pos == t_dirt || ter_at_pos == t_pit ||
               ter_at_pos == t_dirtmound || ter_at_pos == t_pit_shallow ||
-              ter_at_pos == t_grass ) && !vp &&
-            furn_at_pos == f_null ) {
+              ter_at_pos == t_grass || furn_at_pos == furn_str_id( "f_planter" ) ) && !vp &&
```

Comment:
Don't have hardcoded check for a specific terrain id, look for the `PLANTABLE` flag.

Reply:
I've changed it to look for PLANTABLE flag, though I left some hardcoded checks, like pits etc, which don't have PLANTABLE flags, but are reasonable as sleeping surfaces for plant.